src/ssr-template-helper.js

```diff
@@ -113,34 +107,36 @@ export class SsrTemplateHelper {
    */
   buildPayload_(
     request, mustacheTemplate, opt_templates, opt_attributes = {}) {
-    const ampComponent = dict({
-      'type': this.sourceComponent_,
-      'successTemplate': {
+    const ampComponent = dict({'type': this.sourceComponent_});
+    if ((opt_templates && opt_templates['successTemplate'])
+        || mustacheTemplate) {
+      ampComponent['successTemplate'] = {
         'type': 'amp-mustache',
-        'payload': opt_templates
-          ? this.xmls_.serializeToString(opt_templates['successTemplate'])
+        'payload': (opt_templates && opt_templates['successTemplate'])
+          ? opt_templates['successTemplate']./*REVIEW*/innerHTML
           : mustacheTemplate,
```

Comment:
Don't you need to get `innerHTML` of `mustacheTemplate`?

Reply:
Not really, mustacheTemplate only comes from line 87 this file. I called .innerHTML there.

Reply:
I think it's confusing to call `innerHTML` in different functions. Why not pass in the actual template object and get innerHTML in the same function? That way you can just save a `successTemplate` variable and call `innerHTML` once on it.